### PR TITLE
core: vm: implement byzantine reporting mechanism

### PR DESCRIPTION
**Pull request by aitjcize**

### CODE REVIEW

File: core/vm/governance.go [hunk 1]
```diff
@@ -2229,6 +2474,28 @@ func (g *GovernanceContract) addDKGComplaint(round *big.Int, comp []byte) ([]byt
 		return g.penalize()
 	}
 
+	mpk, err := g.state.GetDKGMasterPublicKeyByProposerID(
```
Comment (aitjcize) [inline core/vm/governance.go]: @jimmyhu-dexon can you help check this part,  am I using the correct mpk?

Comment (jimmyhu-dexon) [reply core/vm/governance.go]: Yes, this one is correct.

File: core/vm/governance.go [hunk 2]
```diff
@@ -2229,6 +2475,28 @@ func (g *GovernanceContract) addDKGComplaint(round *big.Int, comp []byte) ([]byt
...
+	}
+
+	// Verify DKG complaint is correct.
+	ok, err := coreUtils.VerifyDKGComplaint(&dkgComplaint, mpk)
+	if !ok || err != nil {
+		return g.penalize()
+	}
+
+	// Fine the attacker.
+	if !dkgComplaint.IsNack() {
```
Comment (jimmyhu-dexon) [inline core/vm/governance.go]: Use `coreUtils.NeedPenaltyDKGPrivateShare` to check if the attacker should be fined.

File: core/vm/governance.go [hunk 3]
```diff
@@ -2639,6 +2907,89 @@ func (g *GovernanceContract) proposeCRS(nextRound *big.Int, signedCRS []byte) ([
 	return g.useGas(0)
 }
 
+type sortBytes [][]byte
+
+func (s sortBytes) Less(i, j int) bool {
+	return bytes.Compare(s[i], s[j]) <= 0
```
Comment (jimmyhu-dexon) [inline core/vm/governance.go]: is `=` necessary?

File: core/vm/governance.go (hunk 2 above)
Comment (aitjcize) [reply core/vm/governance.go]: oops, forgot to call it.

Comment (aitjcize) [reply core/vm/governance.go]: fixed

File: core/vm/governance.go (hunk 3 above)
Comment (aitjcize) [reply core/vm/governance.go]: removed.

**Pull request merged**